ddtrace/tracing/_span_link.py

```diff
@@ -0,0 +1,98 @@
+"""
+Span Links
+==========
+
+Description
+-----------
+
+``ddtrace.trace.SpanLink`` introduces a new causal relationship between spans.
+This new behavior is analogous to OpenTelemetry Span Links:
+https://opentelemetry.io/docs/concepts/signals/traces/#span-links
+
+
+Usage
+-----
+
+SpanLinks can be set using :meth:`ddtrace.Span.link_span(...)` Ex::
+
+    from ddtrace import tracer
+
+    s1 = tracer.trace("s1")
+    s2 = tracer.trace("s2")
+
+    link_attributes = {"link.name": "s1_to_s2", "link.kind": "scheduled_by", "key1": "val1"}
+    s1.link_span(s2.context, link_attributes)
+"""
+
+from typing import Optional
+
+import attr
+
+
+@attr.s
+class SpanLink:
+    """
+    TraceId [required]: The span's 128-bit Trace ID
+    SpanId [required]: The span's 64-bit Span ID
+    TraceFlags [optional]: The span's trace-flags field, as defined in the W3C standard
+
+    This field must be omitted if no sampling information was provided. If the original
+    OpenTelemetry flags are known, they should be directly provided here. If only sampling
+    information is provided, the flags value must be 1 if the decision is keep, otherwise 0.
+
+    TraceState [optional]: The span's tracestate field, as defined in the W3C standard
+    If datadog (or b3) headers are given, they must be transformed to an equivalent tracestate
+    representation (see [RFC] Implementing W3C Trace Context Propagation).
+
+    Attributes: Zero or more key-value pairs, which have the following properties
+    The key must be a non-empty string
+    The value is either:
+    A primitive type: string, bool, or number
+    An array of primitive type values
+    Optional: An array of arrays.
+    """
```

Comment:
```suggestion
    """
    TraceId [required]: The span's 128-bit Trace ID
    SpanId [required]: The span's 64-bit Span ID

    Flags [optional]: The span's trace-flags field, as defined in the W3C standard. If only sampling
    information is provided, the flags value must be 1 if the decision is keep, otherwise 0.

    TraceState [optional]: The span's tracestate field, as defined in the W3C standard.

    Attributes [optional]: Zero or more key-value pairs, where the key must be a non-empty string and the
    value is either a string, bool, number or an array of primitive type values.
    """
```